apps/prairielearn/src/ee/lib/ai-grading.ts

```diff
@@ -44,34 +46,76 @@ const GradedExampleSchema = z.object({
   submission_text: z.string(),
   score_perc: z.number(),
   instance_question_id: z.string(),
+  manual_rubric_grading_id: z.string().nullable(),
 });
 type GradedExample = z.infer<typeof GradedExampleSchema>;
 
-async function generateGPTPromptWithExamples({
+const AppliedRubricItemSchema = z.object({
+  rubric_item_id: IdSchema,
+  score: z.coerce.number().nullish(),
+});
+type AppliedRubricItem = z.infer<typeof AppliedRubricItemSchema>;
+function parseRubricItems({
+  rubric_items,
+  gpt_rubric_items,
+}: {
+  rubric_items: RubricItem[];
+  gpt_rubric_items: number[];
+}): AppliedRubricItem[] {
+  const output: AppliedRubricItem[] = [];
+  for (const rubric_item_number of gpt_rubric_items) {
+    output.push(
+      AppliedRubricItemSchema.parse({ rubric_item_id: rubric_items[rubric_item_number].id }),
+    );
+  }
+  return output;
+}
+
+async function generateGPTPrompt({
   question_prompt,
   student_answer,
   example_submissions,
+  rubric_items,
 }: {
   question_prompt: string;
   student_answer: string;
   example_submissions: GradedExample[];
-}): Promise<
-  {
+  rubric_items: RubricItem[];
+}): Promise<{
+  messages: {
     role: 'system' | 'user';
     content: string;
-  }[]
-> {
+  }[];
+  warning: string;
+}> {
   const messages: {
     role: 'system' | 'user';
     content: string;
   }[] = [];
+  let warning = '';
```

Comment:
I think we want to just directly call `job.warn` on the `job` to do all the warnings, no need to accumulate them in a separate string.

Reply:
I'm currently adding one extra sentence of explanation to the warning when the warning is not none, so it made sense to me to define warning as a variable so I can check it later